src/compute/src/rpc/service/monitor_service.rs

```diff
@@ -77,12 +77,23 @@ impl MonitorService for MonitorServiceImpl {
         &self,
         request: Request<StackTraceRequest>,
     ) -> Result<Response<StackTraceResponse>, Status> {
-        let _req = request.into_inner();
+        let req = request.into_inner();
 
         let actor_traces = if let Some(reg) = self.stream_mgr.await_tree_reg() {
             reg.collect::<Actor>()
                 .into_iter()
-                .map(|(k, v)| (k.0, v.to_string()))
+                .map(|(k, v)| {
+                    (
+                        k.0,
+                        if req.actor_traces_format.is_some()
+                            && req.actor_traces_format.as_ref().unwrap() == "text"
+                        {
+                            v.to_string()
+                        } else {
+                            serde_json::to_string(&v).unwrap()
+                        },
```

Comment:
The logic is different from the comment: "// Allowed values: "json", "text". Default is "text"."

Here it defaults to `json`.

Maybe better to define an enum instead of using string

Reply:
this is not resolved

Reply:
Updated.